### PR TITLE
Update template sample. Fixes #729.

### LINKED ISSUE #729: Docs: code sample should include properties getter
In this section:

https://lit-element.polymer-project.org/guide/templates#design-a-performant-template

The inline code sample doesn't show the `properties` getter. That's important to the sample (since behavior depends on whether or not `message` is defined in `properties`).

### PR DESCRIPTION
**Pull request by arthurevans**
Added a properties getter to this example as suggested in #729. 

Also fixed several examples with broken syntax.

Oh, yeah. Also deleted one example project that we weren't using anywhere.

### CODE REVIEW

File: docs/_includes/projects/docs/templates/databinding/my-element.js [hunk 1]
```diff
@@ -3,10 +3,10 @@ import { LitElement, html } from 'lit-element';
 class MyElement extends LitElement {
   static get properties() {
     return {
-      prop1: String,
-      prop2: String,
-      prop3: Boolean,
-      prop4: String
+      prop1: {type: String},
```
Comment (justinfagnani) [inline docs/_includes/projects/docs/templates/databinding/my-element.js:6]: How about only including one `{type: string}` and make the others `{}` with a `// type: string is the default` comment?

Comment (arthurevans) [reply docs/_includes/projects/docs/templates/databinding/my-element.js:6]: I think if we're going to show an empty declaration, it should probably be in the Properties chapter, not here.

I wasn't sure whether we should consider `{}` an anti-pattern or a convenient shorthand.

Comment (justinfagnani) [reply docs/_includes/projects/docs/templates/databinding/my-element.js:6]: `{}` in JS is the same as `@property()` in TS. It's perfectly fine.

Comment (arthurevans) [reply docs/_includes/projects/docs/templates/databinding/my-element.js:6]: OK. I'm going to try to rationalize these samples in a subsequent PR. Just wanted to unbreak them here.

**Pull request merged**